test/task-affinity.jl

```diff
@@ -0,0 +1,577 @@
+@testset "Dagger Task affinity Tests" begin
+
+    function shuffle!(v::AbstractVector)
+        for i in length(v):-1:2
+            j = rand(1:i)
+            v[i], v[j] = v[j], v[i]
+        end
+        return v
+    end
+
+    get_compute_scope(x::DTask) = try
+        Dagger.Sch._find_thunk(x).compute_scope
+    catch
+        Dagger.InvalidScope
+    end
+
+    get_result_scope(x::DTask) = try
+        fetch(x; raw=true).scope
+    catch
+        Dagger.InvalidScope
+    end
+
+    get_execution_scope(x::DTask) = try
+        chunk = fetch(x; raw=true)
+        Dagger.ExactScope(chunk.processor)
+    catch
+        Dagger.InvalidScope
+    end
+
+    function intersect_scopes(scope1::Dagger.AbstractScope, scopes::Dagger.AbstractScope...)
+        for s in scopes
+            scope1 = Dagger.constrain(scope1, s)
+            scope1 isa Dagger.InvalidScope && return (scope1,)
+        end
+        return scope1.scopes
+    end
+
+    availprocs  = [proc for i in Dagger.procs() for proc in Dagger.get_processors(Dagger.OSProc(i))]
+    availscopes = shuffle!(Dagger.ExactScope.(availprocs))
+    numscopes   = length(availscopes)
+
+    master_proc  = Dagger.ThreadProc(1, 1)
+    master_scope = Dagger.ExactScope(master_proc)
+
+    @testset "Dagger Function: scope, compute_scope and result_scope Tests" begin
+
+        @everywhere f(x) = x + 1
+
+        @testset "scope Tests" begin
+            scope_only = Dagger.UnionScope(rand(availscopes, rand(1:length(availscopes))))
+
+            task1 = Dagger.@spawn scope=scope_only f(10); wait(task1); fetch(task1)
+            @test get_compute_scope(task1) == scope_only
+            @test get_result_scope(task1) == Dagger.AnyScope()
+
+            execution_scope1 = get_execution_scope(task1)
+            @test execution_scope1 in intersect_scopes(execution_scope1,scope_only)
+        end
+
+        @testset "compute_scope Tests" begin
+            compute_scope_only = Dagger.UnionScope(rand(availscopes, rand(1:length(availscopes))))
+            scope              = Dagger.UnionScope(rand(availscopes, rand(1:length(availscopes))))
+
+            task1 = Dagger.@spawn compute_scope=compute_scope_only f(10);             wait(task1); fetch(task1)
```

Comment:
```suggestion
            task1 = Dagger.@spawn compute_scope=compute_scope_only f(10);             fetch(task1)
```

`fetch` already calls `wait` internally, so you don't need both.